src/metabase/query_processor/pivot.clj

```diff
@@ -357,13 +357,14 @@
         metadata-provider  (or (:lib/metadata query)
                                (lib.metadata.jvm/application-database-metadata-provider (:database query)))
         query              (lib/query metadata-provider query)
-        index-in-breakouts (into {}
-                                 (comp (filter (comp #{:source/breakouts} :lib/source))
-                                       (map-indexed (fn [i column] [(:name column) i])))
-                                 (lib/returned-columns query))
+        returned-columns   (lib/returned-columns query)
+        column-name->index (into {}
+                                 (map-indexed (fn [i column] [(:name column) i]))
+                                 (concat (filter (comp #{:source/breakouts} :lib/source) returned-columns)
+                                         (filter (comp #{:source/aggregations} :lib/source) returned-columns)))
```

Comment:
(Assuming the order matters here, which makes sense since we order the columns as breakouts and then aggregations.)

```suggestion
        returned-columns   (lib/returned-columns query)
        {:source/keys [aggregations breakouts]} (group-by :lib/source returned-columns)
        column-name->index (into {}
                                 (map-indexed (fn [i column] [(:name column) i]))
                                 (concat breakouts aggregations))
```